assets/cluster-monitoring-operator/monitoring-targets-edit-cluster-role.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: monitoring-targets-edit
+rules:
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  - podmonitors
```

Comment:
according to the OEP we need to have `PrometheusRule` here too.

Reply:
on an after-thought: I believe we added PrometheusRule too much in the enhancement proposal, because as per https://issues.redhat.com/browse/MON-936?focusedCommentId=13977634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13977634 this refers only to ServiceMonitor/PodMonitor custom resources. We should submit a PR against the enhancement.

Reply:
This is what I meant. I think PrometheusRule _should_ be part of this, although then just "targets" doesn't make sense.

Reply:
We can unify `monitoring-rules-edit` and `monitoring-targets-edit`, but that would imply that the user cannot be given distinct permissions to silence alerts.

Reply:
People can still create those roles if they need but I don't see a standard use case where the distinction needs to be present. The basic use case is self service monitoring, and a user will always need both for that.

Reply:
I agree, I don't see a use case for when often one would scrape their metrics but not allowed to add rules for those metrics. Admins can still create those if they need granularity. Or was this a specific requirement from some SRE folks?

Reply:
i am fine with that, let's just make sure @sichvoge is on board too.

Reply:
No strong feelings from my side. It's more a QoL feature anyways and I am ok to combine the two edit roles into one, and fall back to an admin to create more granularity.

Reply:
I've added a comment to the Jira ticket for Christian.

Reply:
@simonpasquier we reached consensus that we need to add `prometheusrule` custom resources here. The OEP doesn't need an update as it specifies that already.

Reply:
I think there is a small change that we should do though, which is change the naming, as this doesn't reflect just targets. How about just `monitoring-config-edit`?

Reply:
I agree about changing the name. The OEP already provisions a [monitoring-workload-config-edit](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/user-workload-monitoring.md#monitoring-workload-config-edit) role so having `monitoring-config-edit` might be confusing?
Maybe simply `monitoring-edit`?

Reply:
I'm happy with `monitoring-edit` as well.